src/tensor_ops/conv2d/cuda_kernel.rs

```diff
@@ -102,13 +102,14 @@ where
         _: &Tensor<O, E, Self>,
         grad_out: &Self::Vec<E>,
     ) -> Result<(), Self::Err> {
-        let patches_numel = op.batch * op.chan_out * op.kernel * op.kernel * op.h_in * op.w_in;
+        let image_numel = op.batch * op.chan_out * op.h_in * op.w_in;
+        let patches_numel = op.kernel * op.kernel * image_numel;
         let mut patches = self.dev.alloc_zeros::<E>(patches_numel)?;
```

Comment:
Any thoughts on how to get rid of this memset operation here? I tried changing this to `self.dev.alloc::<E>`, which just allocates but doesn't do the memset, but the unfolding kernel doesn't necessarily set every position if its launched on the image pixels as you do below.

If we can get rid of the memsets on the patches its a pretty big improvement too

Reply:
I changed all the alloc_zeros to alloc except for `grad_f_b1023`, and it didnt seem to make a difference (82.5s alloc vs 86.6s alloc_zeros, which is within variance). I'm guessing memset is run async (or its just fast) and so it doesn't actually affect the runtime much.